src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -1,765 +1,774 @@
-    {
-        if (auto func = reinterpret_cast<GetDpiForMonitorInternalFunc>(GetProcAddress(user32.get(), "GetDpiForMonitorInternal")))
-        {
-            func(m_monitor, 0, &dpi, &dpi);
-        }
-    }
-
-    if (dpi == 0)
-    {
-        if (wil::unique_hdc hdc{ GetDC(nullptr) })
-        {
-            dpi = GetDeviceCaps(hdc.get(), LOGPIXELSX);
-        }
-    }
-
-    return (dpi == 0) ? DPIAware::DEFAULT_DPI : dpi;
-}
-#pragma endregion
-
-LRESULT CALLBACK ZoneWindow::s_WndProc(HWND window, UINT message, WPARAM wparam, LPARAM lparam) noexcept
-{
-    auto thisRef = reinterpret_cast<ZoneWindow*>(GetWindowLongPtr(window, GWLP_USERDATA));
-    if ((thisRef == nullptr) && (message == WM_CREATE))
-    {
-        auto createStruct = reinterpret_cast<LPCREATESTRUCT>(lparam);
-        thisRef = reinterpret_cast<ZoneWindow*>(createStruct->lpCreateParams);
-        SetWindowLongPtr(window, GWLP_USERDATA, reinterpret_cast<LONG_PTR>(thisRef));
-    }
-
-    return (thisRef != nullptr) ? thisRef->WndProc(message, wparam, lparam) :
-        DefWindowProc(window, message, wparam, lparam);
-}
-
-winrt::com_ptr<IZoneWindow> MakeZoneWindow(IZoneWindowHost* host, HINSTANCE hinstance, HMONITOR monitor,
-    PCWSTR deviceId, PCWSTR virtualDesktopId, bool flashZones) noexcept
-{
-    return winrt::make_self<ZoneWindow>(host, hinstance, monitor, deviceId, virtualDesktopId, flashZones);
-}
+        }
+    }
+
+    if ((m_keyCycle > 0) && loopAround)
+    {
+        // Cycling through a non-empty group and hit the end
+        m_keyCycle = 0;
+        OnKeyUp(wparam);
+    }
+    else
+    {
+        m_keyCycle++;
+    }
+
+    if (m_host)
+    {
+        m_host->MoveWindowsOnActiveZoneSetChange();
+    }
+    m_highlightZone = nullptr;
+}
+
+void ZoneWindow::FlashZones() noexcept
+{
+    m_flashMode = true;
+
+    ShowWindow(m_window.get(), SW_SHOWNA);
+    std::thread([window = m_window.get()]() {
+        AnimateWindow(window, m_flashDuration, AW_HIDE | AW_BLEND);
+    }).detach();
+}
+#pragma endregion
+
+LRESULT CALLBACK ZoneWindow::s_WndProc(HWND window, UINT message, WPARAM wparam, LPARAM lparam) noexcept
+{
+    auto thisRef = reinterpret_cast<ZoneWindow*>(GetWindowLongPtr(window, GWLP_USERDATA));
+    if ((thisRef == nullptr) && (message == WM_CREATE))
+    {
+        auto createStruct = reinterpret_cast<LPCREATESTRUCT>(lparam);
+        thisRef = reinterpret_cast<ZoneWindow*>(createStruct->lpCreateParams);
+        SetWindowLongPtr(window, GWLP_USERDATA, reinterpret_cast<LONG_PTR>(thisRef));
+    }
+
+    return (thisRef != nullptr) ? thisRef->WndProc(message, wparam, lparam) :
+                                  DefWindowProc(window, message, wparam, lparam);
+}
+
+winrt::com_ptr<IZoneWindow> MakeZoneWindow(IZoneWindowHost* host, HINSTANCE hinstance, HMONITOR monitor, std::wstring uniqueId, bool flashZones) noexcept
+{
+    auto self = winrt::make_self<ZoneWindow>(hinstance);
+    if (self->Init(host, hinstance, monitor, uniqueId, flashZones))

```

Comment:
nit: let's `std::move(uniqueId)`

Reply:
Maybe we can use const reference to unique id instead of copying `std::wstring` through function calls ?

Reply:
yep, that also works!